tests/unit/plugins/module_utils/test_client.py

```diff
@@ -71,8 +71,18 @@ def test_json_is_cached(self, mocker):
 
 
 class TestClientAuthHeader:
+    def test_basic_auth_no_host(self):
+        msg = "Invalid instance host value: ''. Value must start with 'https://' or 'http://'"
+        with pytest.raises(errors.ServiceNowError, match=msg):
+            client.Client("", "user", "pass")
+
+    def test_basic_auth_host_without_schema(self):
+        msg = "Invalid instance host value: 'instance.com'. Value must start with 'https://' or 'http://'"
+        with pytest.raises(errors.ServiceNowError, match=msg):
+            client.Client("instance.com", "user", "pass")
```

Comment:
This code does not test all scenarios that the code can actually encounter while executing. We should be, at minimum, also testing the `host=None` combination since this is something we can get when loading data from environment variables in the inventory plugin.

Also, since we are actually testing the same invariant using different input data, we should combine all tests into a single one using pytest's parameterized marker. Something like this should work:

```
    @pytest.mark.parametrize("host", [None, "", "invalid", "missing.schema"])
    def test_invalid_host(self, host):
        with pytest.raises(errors.ServiceNowError, match="Invalid instance host value"):
            client.Client(host, "user", "pass")
```

The code also fails to test the happy path. I know we do test it indirectly in other tests, but having an explicit test will make clear what we expect from the `_init__` method:

```
    @pytest.mark.parametrize("host", ["http://insecure.host", "https://secure.host"])
    def test_invalid_host(self, host):
        client.Client(host, "user", "pass")
```